A6.md

```diff
@@ -0,0 +1,398 @@
+
+#### Disabling Retries
+
+Clients cannot override retry policy set by the service config. However, retry support can be disabled entirely within the gRPC client library. This is designed to enable existing libraries that wrap gRPC with their own retry implementation (such as Veneer Toolkit) to avoid having retries taking place at the gRPC layer and within their own libraries.
+
+Eventually, retry logic should be taken out of the wrapping libraries, and only exist in gRPC. But allowing the retries to be disabled entirely will make that rollout process easier.
+
+#### Retry and Hedging Statistics
+
+gRPC will treat each retry attempt or hedged RPC as a distinct RPC with regards to the current per-RPC metrics. For example, when an RPC fails with a retryable status code and a retry attempt is made, the original request and the retry attempt will be recorded as two separate RPCs.
+
+Additionally, to present a clearer picture of retry attempts, we add three additional per-method metrics:
+
+1. Total number of retry attempts made
+2. Total number of retry attempts which failed
+3. A histogram of retry attempts made:
+  1. The number of retry attempts will be classified into the following buckets: 
+    1. >=1, >=2, >=3, >=4, >=5, >=10, >=100, >=1000
+  2. Each retry attempt increments the count in exactly bucket. For example:
+    1. The 1st retry attempt adds one to the ">=1" count
+    2. The 2nd retry attempt adds one to the ">=2" count, leaving the count for ">=1" unchanged.
+    3. The 5th through 9th retry attempts each add one to the ">=5" count.
+    4. The 10th through 99th retry attempts each add one to the ">=10" count.
+
+For hedged requests, we record the same stats as above, treating the first hedged request as the initial RPC and subsequent hedged requests as retry attempts.
+
+### Configuration Language
+
+Retry and hedging configuration is set as part of the service config, which is transmitted to the client during DNS resolution. Like other aspects of the service config, retry or hedging policies can be specified per-method, per-service, or per-server name.
+
+Service owners must choose between a retry policy or a hedging policy. Unless the service owner specifies a policy in the configuration, retries and hedging will not be enabled. The retry policy and hedging policy each have their own set of configuration options, detailed below. 
+
+The parameters for throttling retry attempts and hedged RPCs when failures exceed a certain threshold are also set in the service config. Throttling applies across methods and services on a particular server, and thus may only be configured per-server name.
+
+#### Retry Policy
+
+This is an example of a retry policy and its associated configuration. It implements exponential backoff with a maximum of three retry attempts, only retrying RPCs when an `UNAVAILABLE` status code is received.
+
+```
+'retry_policy': {
```

Comment:
@lukaszx0, can you please chime in on the comment thread for #5 (https://groups.google.com/d/topic/grpc-io/DkweyrWEXxU/discussion) and describe the limitations that would prevent you from using DNS for service configs?  I'd like more information on that, but I'd like to keep this discussion in that other thread, since it's not really directly related to the retry design.  Thanks!